docs/external-code.md

```diff
@@ -0,0 +1,131 @@
+
+```lua title=things.lua
+return {
+    {
+        foo = 123,
+        bar = 456,
+    },
+    {
+        foo = 789,
+        bar = 987,
+    },
+}
+```
+
+Writing a definitions file for this is tricky, since the Lua file has no named imports and no default export. Here, you have to use `export =` syntax, like so:
+
+```ts title=things.d.ts
+interface Thing {
+  foo: number;
+  bar: number;
+}
+
+declare const things: Thing[];
+export = things;
+```
+
+Then, in your TypeScript code, you can import it like:
+
+```ts title=main.ts
+import * as things from "./module";
+
+print(things[0].foo); // Prints "123"
+```
+
+For more information about this export syntax, see [the official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require).
+
+## Importing and creating Lua packages from npm
+
+TSTL supports module resolution for libraries, which means you can _use_ and _create_ npm packages containing `.lua` files. (Most packages on npm contain JavaScript files, but npm allows you to create packages with whatever kinds of files you want.)
+
+### Using Lua packages
+
+To use a Lua package, install it via npm and use it in the same way that you would in a normal TypeScript project. In other words:
+
+```sh
+# If you use npm:
+npm install foo --save
+
+# If you use yarn:
+yarn add foo
+
+# If you use pnpm
+pnpm add foo
+```
+
+And then use it in your code:
+
+```ts
+import { someFunction } from "foo";
+
+someFunction();
+```
+
+Since the npm package was presumably made for TSTL users, it will almost certainly include `.d.ts` files alongside the `.lua` files, which is necessary for TSTL to import the Lua files properly. If there are no `.d.ts` files, you can try [creating some for the package yourself](./advanced/writing-declarations.md).
+
+### Creating Lua packages
+
+Right now, there are not very many Lua/TSTL packages on npm. If you do publish one, put the keywords of "typescript-to-lua", "lua", and "tstl" in your "package.json" file so that the package is searchable.
+
+For an example of a Lua/TSTL package published to npm, see [`isaacscript-common`](https://github.com/IsaacScript/isaacscript/tree/main/packages/isaacscript-common).
+
+## Importing JavaScript or TypeScript packages from npm
+
+**Importing JavaScript or TypeScript packages from npm will not work.** This means that it is impossible to use common JavaScript/TypeScript libraries like [Underscore.js](https://underscorejs.org/) and so on.
+
+(It is not possible for TSTL to work with a generic npm package because most TypeScript libraries only publish the compiled JavaScript to npm. And TSTL can't convert JavaScript to Lua because it needs the type information to create correct code. For example, TSTL needs to be able to distinguish between arrays and objects in order to write the correct index.)
+
+As a workaround, you can copy paste TypeScript code from an npm package directly into your project.
```

Comment:
```suggestion
As a workaround, you can copy paste TypeScript code from a package repository into your project, such that it will be included in the TSTL build.
```